fe/fe-core/src/main/java/com/starrocks/http/rest/RestBaseAction.java

```diff
@@ -150,6 +150,9 @@ public void execute(BaseRequest request, BaseResponse response) throws DdlExcept
         ctx.setCurrentUserIdentity(currentUser);
         ctx.setCurrentRoleIds(currentUser);
         ctx.setThreadLocalInfo();
+        if (request.getRequest().headers().contains(WAREHOUSE_KEY)) {
```

Comment:
please add some unit tests about the new parameters, and what's the response looks like if the specified warehouse doesn't exist.

Reply:
I added a UT to post a query_plan request, just like sr flink source connector:
https://github.com/StarRocks/starrocks-connector-for-apache-flink/blob/3e17b7b0531f128fb722b4892fdf608931854202/src/main/java/com/starrocks/connector/flink/manager/StarRocksQueryPlanVisitor.java#L95-#L102